### ISSUE TITLE
Refresh Modal component

### ISSUE THREAD

**Issue by connor-baer** [sumup-oss/circuit-ui#791]
## Component to amend

👀 [View Modal, ModalContext, ModalHeader, and ModalFooter in Storybook](https://circuit.sumup.com/?path=/story/components-modal--base)

### API

- Deprecate the Modal component. It shouldn't be used directly and should have never been publicly exposed.
- Expose the ModalContext as the only way to open modals. It takes care of important side effects like body scroll locking and can keep track of how many modals are open, so we can increase the box-shadow and other stacking depths related things (see iOS for inspiration) → need to check with designers if they want stacking
- Combine the modal subcomponents (header, footer, etc.) into one modal component. The header can be turned into a `title` prop, the footer into a `footer` or `actions` prop. This gives us greater control over how we display them and allows us to place the a Suspense boundary around the content (and footer?), but not the header.
